packages/deep-merge/src/deep-merge.ts

```diff
@@ -64,11 +95,25 @@ function cloneObjectProperties(target: any, [key, value]: any): any {
   return target;
 }
 
-function isObject(obj: any): obj is object {
-  return obj && typeof obj === 'object' && !Array.isArray(obj);
+/**
+ * Returns true if the revieved parameter is an object.
+ *
+ * @param obj - The object to check.
+ *
+ * @returns A boolean indicating if the parameter is an object.
+ */
+function isObject(obj: any): obj is Record<string, unknown> {
+  return obj && typeof obj === "object" && !Array.isArray(obj);
 }
 
-function mergeObject(target: any, [key, value]: any): any {
+/**
+ * A function which merges two sources.
+ *
+ * @param target - The target object to clone in.
+ * @param source - The source objet to clone.
```

Comment:
This is not the `source` object, but an entry (key, value) to clone into the `target` object